sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala

```diff
@@ -636,7 +644,12 @@ abstract class CastBase extends UnaryExpression with TimeZoneAwareExpression wit
         // Please refer to https://github.com/apache/spark/pull/26640
         changePrecision(Decimal(new JavaBigDecimal(s.toString.trim)), target)
       } catch {
-        case _: NumberFormatException => null
+        case _: NumberFormatException =>
+          if (ansiEnabled) {
+            throw new NumberFormatException(s"invalid input syntax for type numeric: $s")
```

Comment:
can we catch and re-throw `NumberFormatException` with better error message in `changePrecision`? Then here can be simple
```
case _: NumberFormatException if !ansiEnabled => null
```

Reply:
@cloud-fan `NumberFormatException` is thrown due to `new JavaBigDecimal(s.toString.trim)`. 
And this can't be handled in `changePrecision` as its taking a decimal value as input. So i think its better to handle this exception here only.

Reply:
makes sense